GMX-Bots/Large-Open-and-Increase-Positions/src/agent.spec.ts

```diff
@@ -0,0 +1,549 @@
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should detect finding for IncreasePosition event whose size is large", async () => {
+    const positionSize = ethers.BigNumber.from(600000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+    const positionSizeDelta = ethers.BigNumber.from(500000000).mul(
+      ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3)
+    );
+    const increasePositionEventLog = MOCK_IFACE.encodeEventLog(increasePositionEventFragment, [
+      key,
+      MOCK_ACCOUNT,
+      MOCK_TOKEN,
+      MOCK_TOKEN,
+      0,
+      positionSizeDelta,
+      true,
+      0,
+      0,
+    ]);
+
+    const updatePositionEventLog = MOCK_IFACE.encodeEventLog(updatePositionEventFragment, [
+      key,
+      positionSize,
+      0,
+      0,
+      0,
+      0,
+      0,
+    ]);
+
+    txEvent = new TestTransactionEvent()
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        increasePositionEventLog.data,
+        ...increasePositionEventLog.topics
+      )
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        updatePositionEventLog.data,
+        ...updatePositionEventLog.topics
+      );
+
+    findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([
+      mockCreateFinding(
+        MOCK_ACCOUNT,
+        mockNetworkManager.vaultAddress,
+        key,
+        key,
+        positionSize,
+        positionSizeDelta,
+        ethers.BigNumber.from(positionSize).sub(ethers.BigNumber.from(positionSizeDelta))
+      ),
+    ]);
+  });
+
+  it("should detect finding for UpdatePosition event whose position size is large", async () => {
+    const increasePositionEventLog = MOCK_IFACE.encodeEventLog(increasePositionEventFragment, [
+      key,
+      MOCK_ACCOUNT,
+      MOCK_TOKEN,
+      MOCK_TOKEN,
+      0,
+      sizeDelta4,
+      true,
+      0,
+      0,
+    ]);
+
+    const updatePositionEventLog = MOCK_IFACE.encodeEventLog(updatePositionEventFragment, [
+      key,
+      sizeDelta4,
+      0,
+      0,
+      0,
+      0,
+      0,
+    ]);
+
+    txEvent = new TestTransactionEvent()
+
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        increasePositionEventLog.data,
+        ...increasePositionEventLog.topics
+      )
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        updatePositionEventLog.data,
+        ...updatePositionEventLog.topics
+      );
+    findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([
+      mockCreateFinding(
+        MOCK_ACCOUNT,
+        mockNetworkManager.vaultAddress,
+        key,
+        key,
+        size4,
+        sizeDelta4,
+        ethers.BigNumber.from(size4).sub(ethers.BigNumber.from(sizeDelta4))
+      ),
+    ]);
+  });
+
+  it("should detect multiple findings for both IncreasePosition and UpdatePosition events with large positions", async () => {
```

Comment:
I don't understand why you separated the test cases in two different transactions. The goal of this test case is making sure the bot still works well when we have the monitored behaviour multiple times.

This way you're never testing the bot when there's both a large increase and a large position opening in the same transaction. 

Let me know if there's a reason why you decided to make this separation, otherwise I recommend having all the events in the same transaction.